### PR TITLE
Create custom type for callbacks

### PR DESCRIPTION
**Pull request by NathanaelGT**
This PR is a implementation from this proposal https://github.com/filamentphp/filament/discussions/2639#discussioncomment-3104926

### CODE REVIEW

File: packages/forms/src/Callbacks/Callback.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Filament\Forms\Callbacks;
```
Comment (danharrin) [inline packages/forms/src/Callbacks/Callback.php]: Potentially we should just use `Filament\Forms` instead. Thoughts?

File: packages/forms/src/Callbacks/Callback.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+use Filament\Forms\Components\Component;
+
+abstract class Callback
+{
+    public function __construct(
```
Comment (danharrin) [inline packages/forms/src/Callbacks/Callback.php]: Also I think this Callback class abstraction is too far, we can just put this constructor within Get and Set individually

File: packages/forms/src/Callbacks/Callback.php (hunk 1 above)
Comment (NathanaelGT) [reply packages/forms/src/Callbacks/Callback.php]: Honestly I'm not sure about the best namespace for this

**Comment by danharrin** [comment]
Thanks

**Pull request merged**